lib/api/controller/auth.js

```diff
@@ -326,7 +365,14 @@ class AuthController extends NativeController {
    * @returns {Promise<object>}
    */
   async checkToken (request) {
-    const token = this.getBodyString(request, 'token');
+    let token = null;
+
+    const isCookieOnly = request.input.args.cookieOnly;
+    if (!global.kuzzle.config.security.supportBrowserCookieAsJwt || !isCookieOnly) {
+      token = this.getBodyString(request, 'token');
+    } else {
+      token = request.context.token.jwt;
+    }
```

Comment:
Nitpicking: coding standards
```suggestion
    } 
    else {
      token = request.context.token.jwt;
    }
```